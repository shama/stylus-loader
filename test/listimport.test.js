import path from 'path';

import {
  compile,
  getCodeFromBundle,
  getCompiler,
  getErrors,
  getWarnings,
} from './helpers';

describe('listimport', () => {
  const cases = [
    {
      name: 'recognizes @import',
      asset: './imports/import.styl',
    },
    {
      name: 'recognizes @require',
      asset: './imports/require.styl',
    },
    {
      name: 'recognizes use()',
      asset: './imports/use.styl',
    },
    {
      name: 'recognizes json()',
      asset: './imports/json.styl',
    },
    {
      name: 'recognizes block-level imports',
      asset: './imports/block-level.styl',
    },
  ];

  for (const item of cases) {
    it(item.name, async () => {
      const testId = item.asset;
      const compiler = getCompiler(
        testId,
        {},
        {
          module: {
            rules: [
              {
                test: /\.styl$/i,
                rules: [
                  {
                    loader: require.resolve('./helpers/testLoader'),
                  },
                  {
                    loader: path.resolve(
                      __dirname,
                      './helpers/stylus-import-loader'
                    ),
                    options: {},
                  },
                ],
              },
            ],
          },
        }
      );
      const stats = await compile(compiler);
      const codeFromBundle = getCodeFromBundle(stats, compiler);

      expect(codeFromBundle.css).toMatchSnapshot('css');
      expect(getWarnings(stats)).toMatchSnapshot('warnings');
      expect(getErrors(stats)).toMatchSnapshot('errors');
    });
  }
});
