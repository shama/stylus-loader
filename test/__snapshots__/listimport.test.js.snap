// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`listimport recognizes @import: css 1`] = `"module.exports = [\\"local-file.styl\\"];"`;

exports[`listimport recognizes @import: errors 1`] = `Array []`;

exports[`listimport recognizes @import: warnings 1`] = `Array []`;

exports[`listimport recognizes @require: css 1`] = `"module.exports = [\\"local-file.styl\\"];"`;

exports[`listimport recognizes @require: errors 1`] = `Array []`;

exports[`listimport recognizes @require: warnings 1`] = `Array []`;

exports[`listimport recognizes block-level imports: css 1`] = `"module.exports = [\\"outside-block.styl\\",\\"inside-block.styl\\",\\"nested-inside-block.styl\\"];"`;

exports[`listimport recognizes block-level imports: errors 1`] = `Array []`;

exports[`listimport recognizes block-level imports: warnings 1`] = `Array []`;

exports[`listimport recognizes json(): css 1`] = `"module.exports = [\\"neat-json.json\\"];"`;

exports[`listimport recognizes json(): errors 1`] = `Array []`;

exports[`listimport recognizes json(): warnings 1`] = `Array []`;

exports[`listimport recognizes use(): css 1`] = `"module.exports = [\\"neat-functions.js\\"];"`;

exports[`listimport recognizes use(): errors 1`] = `Array []`;

exports[`listimport recognizes use(): warnings 1`] = `Array []`;
